agent/app/agent_capability.go

```diff
@@ -193,6 +195,10 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 	// support external firelens config
 	capabilities = agent.appendFirelensConfigCapabilities(capabilities)
 
+	if agent.cfg.GMSACapable {
```

Comment:
This if condition can be just moved to appendGMSACapabilities and in linux, we don't rly need to check the **agent.cfg.GMSACapable**